systemtest/src/test/java/io/strimzi/systemtest/upgrade/OlmUpgradeST.java

```diff
@@ -0,0 +1,258 @@
+        toVersion = toVersion.equals("HEAD") ? Environment.OLM_LATEST_CONTAINER_IMAGE_TAG_DEFAULT : toVersion;
+        assertThat(afterUpgradeVersionOfCo, is(Environment.OLM_APP_BUNDLE_PREFIX + ".v" + toVersion));
+
+        // 10. perform verification of to version
+        performUpgradeVerification(afterUpgradeVersionOfCo);
+
+        // 11. save install-plan to closed-map
+        OlmResource.getClosedMapInstallPlan().put(OlmResource.getNonUsedInstallPlan(), Boolean.TRUE);
+    }
+
+    private void performUpgradeVerification(String version) {
+        LOGGER.info("====================================================================================");
+        LOGGER.info("============== Verification version of CO:" + version);
+        LOGGER.info("====================================================================================");
+
+        // fetch the tag from imageName: docker.io/strimzi/operator:'[latest|0.19.0|0.18.0]'
+        String containerImageTag = kubeClient().getDeployment(kubeClient().getDeploymentNameByPrefix(Constants.STRIMZI_DEPLOYMENT_NAME))
+            .getSpec()
+            .getTemplate()
+            .getMetadata()
+            .getAnnotations()
+            .get("containerImage").split(":")[1];
+
+        LOGGER.info("Image tag of strimzi operator is {}", containerImageTag);
+
+        // NOT (latest image or default substring(1)) for skipping 'v'0.19.0 on the start...
+        // '6.6.6' is the latest version of cluster operator
+        if (!containerImageTag.equals(OLM_LATEST_CONTAINER_IMAGE_TAG) && (!(containerImageTag.equals("latest") || containerImageTag.equals(Environment.OLM_OPERATOR_VERSION_DEFAULT.substring(1))))) {
+            try {
+                File dir = FileUtils.downloadAndUnzip("https://github.com/strimzi/strimzi-kafka-operator/releases/download/" + containerImageTag + "/strimzi-" + containerImageTag + ".zip");
+
+                deployKafkaFromFile(dir, containerImageTag);
+                waitForReadinessOfKafkaCluster();
+
+                KafkaTopicResource.topic(CLUSTER_NAME, topicUpgradeName).done();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        //  this is round only last version (so kafka is not present)
+        } else if (KafkaResource.kafkaClient().inNamespace(namespace).withName(CLUSTER_NAME).get() == null) {
+            KafkaResource.kafkaPersistent(CLUSTER_NAME, 3).done();
+        }
+
+        KafkaBasicExampleClients kafkaBasicClientJob = new KafkaBridgeExampleClients.Builder()
+            .withProducerName(producerName)
+            .withConsumerName(consumerName)
+            .withBootstrapAddress(KafkaResources.plainBootstrapAddress(CLUSTER_NAME))
+            .withTopicName(topicUpgradeName)
+            .withMessageCount(messageUpgradeCount)
+            .withDelayMs(1)
+            .build();
+
+        kafkaBasicClientJob.producerStrimzi().done();
+        kafkaBasicClientJob.consumerStrimzi().done();
```

Comment:
So you actually don't keep clients  attached for whole upgrade, but only for Kafka upgrade. I think you should start clients when the oldest version of CO and kafka is deployed and stop them when all parts of upgrade are done.